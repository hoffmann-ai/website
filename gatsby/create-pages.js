'use strict';

const path = require('path');
const _ = require('lodash');
const createCategoriesPages = require('./pagination/create-categories-pages.js');
const createTagsPages = require('./pagination/create-tags-pages.js');
const createPostsPages = require('./pagination/create-posts-pages.js');

const createPages = async ({ graphql, actions }) => {
  const { createPage } = actions;

  // 404
  createPage({
    path: '/404',
    component: path.resolve('./src/templates/not-found-template.js'),
  });

  // Tags list
  createPage({
    path: '/tags',
    component: path.resolve('./src/templates/tags-list-template.js'),
  });

  // Categories list
  createPage({
    path: '/categories',
    component: path.resolve('./src/templates/categories-list-template.js'),
  });

  const lastPosts = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" } } }
        limit: 4
        sort: { fields: frontmatter___date, order: DESC }
      ) {
        edges {
          node {
            fields {
              slug
            }
            frontmatter {
              date
              description
              title
              category
              socialImage {
                childImageSharp {
                  gatsbyImageData(width: 450, height: 164)
                }
              }
            }
          }
        }
      }
    }
  `);

  const skills = await graphql(`
    {
      allMarkdownRemark(filter: {frontmatter: {template: {eq: "skills"}}}) {
        edges {
          node {
            frontmatter {
              name
              description
              logo {
                childImageSharp {
                  gatsbyImageData(width: 48, height: 48)
                }
              }
            }
          }
        }
      }
    }
  `);

  const clients = await graphql(`
    {
      allFile(filter: { relativeDirectory: { eq: "clients" } }) {
        edges {
          node {
            childImageSharp {
              gatsbyImageData(height: 48)
            }
            name
          }
        }
      }
    }
  `);

  const landing = await graphql(`
    {
      allFile(filter: { relativeDirectory: { eq: "landing" } }) {
        edges {
          node {
            childImageSharp {
              gatsbyImageData(height: 300)
            }
            name
          }
        }
      }
    }
  `);

  const services = await graphql(`
  {
    allMarkdownRemark(filter: {frontmatter: {template: {eq: "services"}}}) {
      edges {
        node {
          frontmatter {
            name
            description
            logo {
              childImageSharp {
                gatsbyImageData(height: 500)
              }
            }
          }
        }
      }
    }
  }
`);

  // Home page
  createPage({
    path: '/',
    component: path.resolve('./src/templates/landing-template.js'),
    context: {
      lastPosts, skills, clients, landing, services
    },
  });

  // Posts and pages from markdown
  const result = await graphql(`
    {
      allMarkdownRemark(filter: { frontmatter: { draft: { ne: true } } }) {
        edges {
          node {
            frontmatter {
              template
            }
            fields {
              slug
            }
          }
        }
      }
    }
  `);

  const { edges } = result.data.allMarkdownRemark;

  _.each(edges, (edge) => {
    if (_.get(edge, 'node.frontmatter.template') === 'page') {
      createPage({
        path: edge.node.fields.slug,
        component: path.resolve('./src/templates/page-template.js'),
        context: { slug: edge.node.fields.slug },
      });
    } else if (_.get(edge, 'node.frontmatter.template') === 'post') {
      createPage({
        path: edge.node.fields.slug,
        component: path.resolve('./src/templates/post-template.js'),
        context: { slug: edge.node.fields.slug },
      });
    }
  });

  // Feeds
  await createTagsPages(graphql, actions);
  await createCategoriesPages(graphql, actions);
  await createPostsPages(graphql, actions);
};

module.exports = createPages;
