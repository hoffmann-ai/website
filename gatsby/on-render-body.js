'use strict';

const React = require('react');

const onRenderBody = ({ setPostBodyComponents }) => {
  setPostBodyComponents([
    <script
      key="axeptio"
      dangerouslySetInnerHTML={{
        __html: `
          window.axeptioSettings = {
            clientId: "60f291c2afa9d02a7f02768e",
            cookiesVersion: "hoffmann.ai-base",
          };
          (function(d, s) {
            var t = d.getElementsByTagName(s)[0], e = d.createElement(s);
            e.async = true;
            e.src = "//static.axept.io/sdk-slim.js";
            t.parentNode.insertBefore(e, t);
          })(document, "script");
      `,
      }}
    />
  ]);
};

module.exports = onRenderBody;
