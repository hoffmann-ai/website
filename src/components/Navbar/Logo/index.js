// @flow strict
export { default } from './Logo';
