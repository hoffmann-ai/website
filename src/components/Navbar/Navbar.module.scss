@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.navbar {
    width: 100%;
    position: -webkit-sticky;
    position: sticky;
    top: 0px;
    padding-top: 10px;
    background-color: rgb(255, 255, 255);
    z-index: 1;
    &__inner {
        &_list {
            display: flex;
            justify-content: space-between;
            align-items: flex-end;
            margin: 0;

            &_item {
                display: inline-block;
            }
        }
    }
}

@include breakpoint-sm {
    .navbar {
        &__inner {
            padding: 0 10px;
        }
    }
}

@include breakpoint-md {
    .navbar {
        &__inner {
            padding: 0;
        }
    }
}
