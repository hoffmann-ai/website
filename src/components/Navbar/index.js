// @flow strict
export { default } from './Navbar';
