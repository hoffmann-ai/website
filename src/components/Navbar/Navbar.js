// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import { GatsbyImage } from 'gatsby-plugin-image';
import Menu from './Menu';
import styles from './Navbar.module.scss';
import Newsletter from '../Newsletter/Newsletter';

import { useGlobalData, useSiteMetadata } from '../../hooks';

const Navbar = () => {
  const { menu } = useSiteMetadata();
  const { logo } = useGlobalData();

  return (
    <div className={styles['navbar']}>
      <nav className={styles['navbar__inner']}>
        <ul className={styles['navbar__inner_list']}>
          <li className={styles['navbar__inner_list_item']}>
            <Link to='/'>
              <GatsbyImage
                className={styles['navbar__inner_list_item_logo']}
                image={logo?.childImageSharp?.gatsbyImageData}
                alt='logo'
              />
            </Link>
          </li>
          <li className={styles['navbar__inner_list_item']}>
            <Menu menu={menu} />
          <Newsletter className={styles['navbar__inner_list_item']} style={'responsive'} />
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default Navbar;
