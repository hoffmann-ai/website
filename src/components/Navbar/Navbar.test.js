// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Navbar from './Navbar';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Navbar', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  const props = {
    isIndex: true
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Navbar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
