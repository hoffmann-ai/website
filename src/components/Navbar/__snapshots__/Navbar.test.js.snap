// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div
  className="navbar"
>
  <nav
    className="navbar__inner"
  >
    <ul
      className="navbar__inner_list"
    >
      <li
        className="navbar__inner_list_item"
      >
        <a
          href="/"
        />
      </li>
      <li
        className="navbar__inner_list_item"
      >
        <nav
          className="menu"
        >
          <ul
            className="menu__list"
          >
            <li
              className="menu__list-item"
            >
              <a
                className="menu__list-item-link"
                href="/test/1/"
              >
                Test label 1
              </a>
            </li>
            <li
              className="menu__list-item"
            >
              <a
                className="menu__list-item-link"
                href="/test/2/"
              >
                Test label 2
              </a>
            </li>
            <li
              className="menu__list-item"
            >
              <a
                className="menu__list-item-link"
                href="/test/3/"
              >
                Test label 3
              </a>
            </li>
          </ul>
        </nav>
      </li>
    </ul>
  </nav>
</div>
`;
